india_banking/public/js/payment_order.js

```diff
@@ -1,363 +1,512 @@
+      (item) => item.payment_status == "Pending"
+    );
+    if (has_pending_payment && frm.doc.docstatus == 1) {
+      frm.add_custom_button(__("Cancel Pending Payments"), function () {
+        show_update_status_dialog(frm);
+      });
+    }
+  },
+
+  set_get_payments_from_buttons(frm) {
+    if (frm.doc.docstatus === 0) {
+      // Define an array of payment sources and their respective triggers
+      const payment_sources = [
+        {
+          label: __("Payment Request"),
+          trigger: "get_payments_from_payment_request",
+        },
+        {
+          label: __("Payment Entry"),
+          trigger: "get_payments_from_payment_entry",
+        },
+        {
+          label: __("Bank Entry(JV)"),
+          trigger: "get_payments_from_journal_entry",
+        },
+      ];
+
+      // Add custom buttons for each payment source
+      payment_sources.forEach((source) => {
+        frm.add_custom_button(
+          source.label,
+          () => frm.trigger(source.trigger),
+          __("Get Payments from")
+        );
+      });
+    }
+  },
+
+  get_payments_from_payment_request(frm) {
+    // Ensure references table is clean before processing
+    frm.trigger("remove_row_if_empty");
+
+    // Collect existing payment requests from references table, if any
+    const existing_payment_requests = (frm.doc.references || []).map(
+      (reference) => reference.payment_request
+    );
+
+    // Use map_current_doc utility to fetch and map payment requests
+    erpnext.utils.map_current_doc({
+      method: "india_banking.overrides.payment_request.make_payment_order",
+      source_doctype: "Payment Request",
+      target: frm,
+      setters: {
+        party_type: "",
+        party: "",
+        grand_total: "",
+      },
+      get_query_filters: {
+        docstatus: 1,
+        status: ["=", "Initiated"],
+        bank: frm.doc.bank,
+        name: ["not in", existing_payment_requests],
+        company: frm.doc.company,
+      },
+    });
+  },
+
+  get_payments_from_payment_entry(frm) {
+    // Ensure references table is clean before processing
+    frm.trigger("remove_row_if_empty");
+
+    // Collect existing payment entries from the references table, if any
+    const existing_payment_entries = (frm.doc.references || []).map(
+      (reference) => reference.payment_entry
+    );
+
+    // Use map_current_doc utility to fetch and map payment entries
+    erpnext.utils.map_current_doc({
+      method: "india_banking.overrides.payment_entry.make_payment_order",
+      source_doctype: "Payment Entry",
+      target: frm,
+      setters: {
+        party: "",
+        paid_amount: "",
+      },
+      get_query_filters: {
+        docstatus: 1,
+        name: ["not in", existing_payment_entries],
+        source_doctype: ["!=", "Payment Request"],
+        payment_type: "Pay",
+        mode_of_payment: "Wire Transfer",
+        bank_account: frm.doc.company_bank_account,
```

Comment:
posting_date lesser than current date, currency being INR, are these conditions taken care of?